tests/crypto/src/common_test.h

```diff
@@ -0,0 +1,497 @@
+	const uint8_t expected_result;                                  /**< Expected result of hmac operation. */
+	const char                      *p_test_vector_name;            /**< Pointer to hmac test vector name. */
+	const char                      *p_input;                       /**< Pointer to input message in hex string format. */
+	const char                      *p_key;                         /**< Pointer to hmac key in hex string format. */
+	const char                      *p_expected_output;             /**< Pointer to expected hmac digest in hex string format. */
+} test_vector_hmac_t;
+
+/**@brief hkdf test vector information.
+ */
+typedef const struct {
+	const u32_t digest_type;                                        /**< Digest type of hkdf operation. */
+	const int expected_err_code;                                    /**< Expected error code from hkdf operation. */
+	const int expected_err_code_expand;                             /**< Expected error code from hkdf expand operation. */
+	const uint8_t expected_result;                                  /**< Expected result of hkdf operation. */
+	const uint8_t expected_result_expand;                           /**< Expected result of hkdf expand operation. */
+	const char                      *p_test_vector_name;            /**< Pointer to hkdf test vector name. */
+	const char                      *p_ikm;                         /**< Pointer to hkdf Input Key Material in hex string format. */
+	const char                      *p_okm;                         /**< Pointer to hkdf Output Key Material in hex string format. */
+	const char                      *p_prk;                         /**< Pointer to hkdf PseudoRandom Key in hex string format. */
+	const char                      *p_salt;                        /**< Pointer to hkdf salt in hex string format. */
+	const char                      *p_info;                        /**< Pointer to hkdf optional application specific information in hex string format. */
+} test_vector_hkdf_t;
+
+const char *get_vector_name(const test_case_t *tc, uint32_t v);
+uint32_t get_vector_count(const test_case_t *tc);
+
+
+
+/**@brief Function for initializing hardware acceleration.
+ *
+ * @details Should only be called once.
+ *
+ * @return								0 on success, else an error code.
+ */
+int init_cc310(void);
+
+/**@brief Function for initializing counter mode deterministic random byte generator.
+ *
+ * @details Should only be called once.
+ *
+ * @param[in]		p_optional_seed		Pointer to a seed. Can be NULL. In that case uses internal seed.
+ * @param[in]      	len					Length of seed in bytes. Ignored if seed is NULL.
+ *
+ * @return								0 on success, else an error code.
+ */
+#if defined(CONFIG_MBEDTLS_CTR_DRBG_ENABLED)
+int init_ctr_drbg(const unsigned char *p_optional_seed, size_t len);
+#endif
+
+/**@brief Function for setting LED_1 to indicate a start of time measurement.
```

Comment:
Is this comment correct? I don't see anything related to led's in the function.

Reply:
Comment was incorrect, updated in 7aee7a9